### PR TITLE
AI functionality in JabRef

### PR DESCRIPTION
**Pull request by InAnYan**
The file name for AI post should be, of course, renamed

### CODE REVIEW

File: _posts/2024-07-01-AI-chatting.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+## AI preferences
+
+Here are some new options in the JabRef preferences.
+
+![AI preferences](../img/AiPreferences.png)
+
+- "Enable AI functionality in JabRef": by default it's turned off, so you need to check this option, if you want to use the new AI features
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face
+- "Chat model": choose the model you like (for OpenAI we recommend `gpt-4o-mini`, as it the cheapest and fastest)
```
Comment (ThiloteE) [inline _posts/2024-07-01-AI-chatting.md:132]: Change "(for OpenAI we recommend `gpt-4o-mini`, as it the cheapest and fastest)" to "(for OpenAI we recommend `gpt-4o-mini`, as to date, it is the cheapest and fastest, although we also recommend to look up the prices from time to time, as they are subject to change)

File: _posts/2024-07-01-AI-chatting.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+*Only 40 grams?* Well, our health is our wealth. And AI helped us to uncover this information!
+
+## How does this work?
+
+In the background, JabRef analyses the linked PDF files of library entries. The information used after the indexing is then supplied to the AI, which, to be precise, in our case is a Large Language Model (LLM). The LLM is currently not stored on your computer. Instead, we have many integrations with AI providers (OpenAI, Mistral AI, Hugging Face), so you can choose the one you like the most. These AI providers are available only remotely via the internet. In short: we send chunks of text to AI service and then receive processed responses. In order to use it you need to configure JabRef to use your [API](https://en.wikipedia.org/wiki/API) key.
+
+## Which AI provider should I use?
+
+We recomend you chosing the OpenAI.
```
Comment (ThiloteE) [inline _posts/2024-07-01-AI-chatting.md:55]: Change to "For now, we recommend you choosing OpenAI"

**Comment by InAnYan** [comment]
Shortened :)

File: _posts/2024-08-12-AI-chatting.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+## End
+
+Thank you for using JabRef and checking out the AI functionality!
+
+This is only the first step of the Google Summer of Code project, so stay tuned for new AI features!
+
+Currently the summarization chatting is under development, so some bugs may occur.
+
+We value your opinion and we want to know: What AI features would you like to see in JabRef in future? What LLM providers would you like to have in next versions?
```
Comment (koppor) [inline _posts/2024-08-12-AI-chatting.md:91]:
```suggestion
We value your opinion and we want to know: What AI features would you like to see in JabRef in future? What LLM providers would you like to have in next versions?
Please show up in our [Feedback forum](https://discourse.jabref.org/c/feedback/3).
```

File: _posts/2024-08-12-AI-chatting.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+(List of LLM providers: [Huggingface](https://huggingface.co/pricing#endpoints), [Azure](https://azure.microsoft.com/en-us/products/#ai-machine-learning), [Together.AI](https://www.together.ai/), [Google](https://ai.google/discover/our-models/), [MistralAI](https://mistral.ai/), [Anthrophic](https://www.anthropic.com/), [Meta](https://llama.meta.com/), ... )
```
Comment (koppor) [inline _posts/2024-08-12-AI-chatting.md:93]: Move this up to "Which AI provider should I use?" - Also explain "..." (maybe, just leave out the dots - or say that OpenAI-API compatible providers can be used.

**Review by koppor — CHANGES_REQUESTED** [review]

File: _posts/2024-08-12-AI-chatting.md (hunk 4 above)
Comment (InAnYan) [reply _posts/2024-08-12-AI-chatting.md:93]: I've changed the text to: "What LLMs or other integrations would you like to have in next versions?". I think this is more known for users rather than AI or LLM provider.

I removed the text. Will that be okay?

File: _posts/2024-08-12-AI-chatting.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+For Mistral AI you need to make a subscription, while for OpenAI you can send money one time.
+
+Hugging Face gives you access to numerous count of models for free. But it will take a very long time for Hugging Face to find a free computer resources for you, and the response time will be also long.
+
+## How to get an API key and what should I do after getting one?
+
+You can refer to [AI user documentation](https://docs.jabref.org/ai/ai-providers-and-api-keys).
+
+## AI preferences
```
[the remark this answers is not in the document]
Comment (InAnYan) [reply _posts/2024-08-12-AI-chatting.md:65]: Oh, I've just seen previous comment (https://github.com/JabRef/blog.jabref.org/pull/93#discussion_r1715411420).

Isn't it good enough?

File: _posts/2024-08-12-AI-chatting.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+## How does this work?
+
+In the background, JabRef analyses the linked PDF files of library entries. The information used after the indexing is then supplied to the AI, which, to be precise, in our case is a Large Language Model (LLM). The LLM is currently not stored on your computer. Instead, we have many integrations with AI providers (OpenAI, Mistral AI, Hugging Face), so you can choose the one you like the most. These AI providers are available only remotely via the internet. In short: we send chunks of text to AI service and then receive processed responses. In order to use it you need to configure JabRef to use your [API](https://en.wikipedia.org/wiki/API) key.
+
+## Which AI provider should I use?
+
+We recomend you chosing the OpenAI.
+
+For Mistral AI you need to make a subscription, while for OpenAI you can send money one time.
```
[the remark this answers is not in the document]
Comment (InAnYan) [reply _posts/2024-08-12-AI-chatting.md:57]: > Mistral AI offers a 8 days test-for-free period?!

Really? I googled, and I haven't found the information for that...

[the remark this answers is not in the document]
Comment (koppor) [reply _posts/2024-08-12-AI-chatting.md:59]: I just signed up and got 5€ credit.

File: _posts/2024-08-12-AI-chatting.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+Hugging Face gives you access to numerous count of models for free.
+
+However, it will take a very long time for Hugging Face to find a free computer resources for you, and the response time will be also long.
+
+## How to get an API key and what should I do after getting one?
+
+Please head to the [AI user documentation](https://docs.jabref.org/ai/ai-providers-and-api-keys) to learn about how to receive a key and where to enter it in the preferences.
```
Comment (koppor) [inline _posts/2024-08-12-AI-chatting.md:67]: The section contains one sentence only. Remove heading and add following sentence before:

In order to use any service, you always need an API key.

File: _posts/2024-08-12-AI-chatting.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+I was wondering, how much chocolate should a human eat in a day? Let's dive in this topic.
+
+![AI second question and answer](../img/AiQuestion2.png)
+
+*Only 40 grams?* Well, our health is our wealth. And AI helped us to uncover this information!
+
+## How does this work?
+
+In the background, JabRef analyses the linked PDF files of library entries. The information used after the indexing is then supplied to the AI, which, to be precise, in our case is a Large Language Model (LLM).
+The LLM is currently not stored on your computer. Instead, we have many integrations with AI providers (OpenAI, Mistral AI, Hugging Face), so you can choose the one you like the most.
```
Comment (koppor) [inline _posts/2024-08-12-AI-chatting.md:52]: You had links to all providers. Can you add them here?

**Review by koppor — CHANGES_REQUESTED** [review]

File: _posts/2024-08-12-AI-chatting.md (hunk 8 above)
Comment (ThiloteE) [reply _posts/2024-08-12-AI-chatting.md:52]: No, not here. If you want them listed, then better in the section down below --> "Which AI provider should I choose". This section here is about "how does this work".

File: _posts/2024-08-12-AI-chatting.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (ThiloteE) [reply _posts/2024-08-12-AI-chatting.md:59]: We can add links, but pricing schemes tend to change over time, so I think we should be conservative here and not go too much into detail.

File: _posts/2024-08-12-AI-chatting.md (hunk 8 above)
Comment (koppor) [reply _posts/2024-08-12-AI-chatting.md:52]: My reviewing "driver": Science: Link at first appearance. 

Maybe the provide list here can be removed. Does it add value to have the concrete list here?

File: _posts/2024-08-12-AI-chatting.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+These AI providers are available only remotely via the internet. In short: we send chunks of text to AI service and then receive processed responses. In order to use it you need to configure JabRef to use your API key.
+
+## Which AI provider should I use?
+
+We recomend you chosing [OpenAI](https://platform.openai.com/docs/models).
+
+For Mistral AI you need to make a subscription, while for OpenAI you can send money one time.
```
Comment (koppor) [inline _posts/2024-08-12-AI-chatting.md:59]: Use "you might need"...

File: _posts/2024-08-12-AI-chatting.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+Hugging Face gives you access to numerous count of models for free.
+
```
Comment (koppor) [inline _posts/2024-08-12-AI-chatting.md:62]: Remove empty line

File: _posts/2024-08-12-AI-chatting.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+![AI second question and answer](../img/AiQuestion2.png)
+
+*Only 40 grams?* Well, our health is our wealth. And AI helped us to uncover this information!
+
+## How does this work?
+
+In the background, JabRef analyses the linked PDF files of library entries. The information used after the indexing is then supplied to the AI, which, to be precise, in our case is a Large Language Model (LLM).
+The LLM is currently not stored on your computer. Instead, we have many integrations with AI providers (OpenAI, Mistral AI, Hugging Face), so you can choose the one you like the most.
+These AI providers are available only remotely via the internet. In short: we send chunks of text to AI service and then receive processed responses. In order to use it you need to configure JabRef to use your API key.
```
Comment (koppor) [inline _posts/2024-08-12-AI-chatting.md:53]: Remove empty line(to join paragraphs). Remove "These AI providers...short:" Then the next sentence "We send..."

I think, this is precise enough

**Review by koppor — CHANGES_REQUESTED** [review]

File: _posts/2024-08-12-AI-chatting.md (hunk 11 above)
Comment (InAnYan) [reply _posts/2024-08-12-AI-chatting.md]: I changed the text. Did I edit everything you proposed?

**Comment by InAnYan** [comment]
Is the blog post okay? Are there any unresolved questions?

**Comment by koppor** [comment]
> Is the blog post okay? Are there any unresolved questions?

One more person should check. I am kind of "blind"...

File: _posts/2024-08-16-AI-chatting.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+## BONUS
+
+Have you ever wanted to try running a local LLM model on your computer with 100% privacy saved? We got you.
+
+We wrote a tutorial of how to install and use `ollama` and integrate it in JabRef.
+
+You find more information at the [user documentation](https://docs.jabref.org/ai/local-llm).
+
+## End
```
Comment (Siedlerchr) [inline _posts/2024-08-16-AI-chatting.md:87]:
```suggestion
## Are you curious?

Try the AI features in our current development version and play around with the new feature 
```

**Comment by Siedlerchr** [comment]
Just one litle addtion but otherwise thumbs up!

File: _posts/2024-08-16-AI-chatting.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option, if you want to use the new AI features
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face
+- "Chat model": choose the model you like (for OpenAI we recommend `gpt-4o-mini`, as to date, it is the cheapest and fastest, although we also recommend to look up the prices from time to time, as they are subject to change)
+- "API token": here you write your API token
+- "Expert settings": here you can change the parameters that affect how AI will generate your answers. If you don't understand the meaning of those settings, don't worry! We have experimented a lot and found the best parameters for you! But if you are curious, then you can refer to [user documentation]()
+
+## BONUS
+
+Have you ever wanted to try running a local LLM model on your computer with 100% privacy saved? We got you.
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:81]: Can we please rephrase this to a more professional sounding wording?

File: _posts/2024-08-16-AI-chatting.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+Here are some new options in the JabRef preferences.
+
+![AI preferences](../img/AiPreferences.png)
+
+- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option, if you want to use the new AI features
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face
+- "Chat model": choose the model you like (for OpenAI we recommend `gpt-4o-mini`, as to date, it is the cheapest and fastest, although we also recommend to look up the prices from time to time, as they are subject to change)
+- "API token": here you write your API token
+- "Expert settings": here you can change the parameters that affect how AI will generate your answers. If you don't understand the meaning of those settings, don't worry! We have experimented a lot and found the best parameters for you! But if you are curious, then you can refer to [user documentation]()
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:77]: Can you please rephrase "If you don't understand ... best parameters for you!" to a more professional sounding wording? Maybe: "The defaults provide good results. In case you would like to modify them, our [user documentation](https://docs.jabref.org/ai/preferences#ai-expert-settings) provides an explanation of it"

(This also adds the link to the user documentation. Currently, the link is empty.)

**Review by koppor — CHANGES_REQUESTED** [review]

File: _posts/2024-08-16-AI-chatting.md [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+tags: [ai, gsoc]
+author: "[Ruslan](https://github.com/InAnYan)"
+---
+
+My name is Ruslan. I’m a novice contributor to JabRef, currently working on an AI project for [Google Summer of Code](https://summerofcode.withgoogle.com/).
+
+I want to introduce you to the new AI features in JabRef.
+
+- AI can generate for you a summary of a research paper
+- You can also chat with papers using a smart AI assistant
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:12]:
```suggestion
My name is Ruslan. I’m a novice contributor to JabRef, currently working on an AI project for [Google Summer of Code](https://summerofcode.withgoogle.com/). My delivered contributions to JabRef is the introduction of two AI features: AI-based summarization of papers and "chatting" with papers using an AI assistant.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+
+## AI summary tab
+
+We have introduced a new entry editor tab, "AI Summary," where AI generates a quick overview of the paper for you.
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:16]:
```suggestion
I introduced a new entry editor tab, "AI Summary," where AI generates a quick overview of the paper for you.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+
+Have you ever considered running a local LLM on your computer while ensuring complete privacy?
+
+We wrote a tutorial of how to install and use `ollama` and integrate it in JabRef.
+
+You find more information at the [user documentation](https://docs.jabref.org/ai/local-llm).
+
+## Are you curious?
+
+Try the AI features in our current development version and explore the new functionality.
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:88]:
```suggestion
Try the AI features in our [current development version](https://builds.jabref.org/main/) and explore the new functionality.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+
+Thank you for using JabRef and checking out the AI functionality!
+
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:91]: This can be removed, can't it?

```suggestion
```

File: _posts/2024-08-16-AI-chatting.md [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+We value your opinion and want to know: What AI features would you like to see in JabRef in the future? What LLMs or other integrations would you like in upcoming versions?
+
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:93]: Merge with the previous paragraph?

```suggestion
```

File: _posts/2024-08-16-AI-chatting.md [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+![AI first question and answer](../img/AiQuestion1.png)
+
+Correct! The authors discuss chocolate components extensively and how it can be a powerful antioxidant.
+
+I was wondering, how much chocolate should a human eat in a day? Let's dive in this topic.
+
+![AI second question and answer](../img/AiQuestion2.png)
+
+*Only 40 grams?* Well, our health is our wealth. And AI helped us to uncover this information!
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:46]:
```suggestion
*Only 40 grams?* Well, our health is our wealth. Exciting that AI helped us to uncover this information!
```

File: _posts/2024-08-16-AI-chatting.md [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+
+## How does this work?
+
+In the background, JabRef analyses the linked PDF files of library entries. After indexing, this information is supplied to the AI, which, in our case, is a Large Language Model (LLM).
+The LLM is currently not stored on your computer. Instead, we have many integrations with AI providers (OpenAI, Mistral AI, Hugging Face), so you can choose the one you like the most.
+We send chunks of text to AI service and then receive processed responses. To use it, you need to configure JabRef with your API key.
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:52]:
```suggestion
JabRef sends chunks of text to AI service and then receive processed responses. To use it, you need to configure JabRef with your API key.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+I was wondering, how much chocolate should a human eat in a day? Let's dive in this topic.
+
+![AI second question and answer](../img/AiQuestion2.png)
+
+*Only 40 grams?* Well, our health is our wealth. And AI helped us to uncover this information!
+
+## How does this work?
+
+In the background, JabRef analyses the linked PDF files of library entries. After indexing, this information is supplied to the AI, which, in our case, is a Large Language Model (LLM).
+The LLM is currently not stored on your computer. Instead, we have many integrations with AI providers (OpenAI, Mistral AI, Hugging Face), so you can choose the one you like the most.
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:51]:
```suggestion
The LLM itself is currently not stored on your computer. Instead, we have many integrations with AI providers (OpenAI, Mistral AI, Hugging Face), so you can choose the one you like the most.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+
+Here are some new options in the JabRef preferences.
+
+![AI preferences](../img/AiPreferences.png)
+
+- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option if you want to use the new AI features
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face
+- "Chat model": choose the model you like (for OpenAI we recommend `gpt-4o-mini`, as to date, it is the cheapest and fastest, though we also recommend to look up the prices periodically, as they are subject to change)
+- "API token": here you write your API token
+- "Expert settings": the defaults provide good results. In case you would like to modify them, our [user documentation](https://docs.jabref.org/ai) provides an explanation of it
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:76]:
```suggestion
- "Expert settings": the defaults provide good results. In case you would like to modify them, our [user documentation](https://docs.jabref.org/ai) provides an explanation of it.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+## AI preferences
+
+Here are some new options in the JabRef preferences.
+
+![AI preferences](../img/AiPreferences.png)
+
+- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option if you want to use the new AI features
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face
+- "Chat model": choose the model you like (for OpenAI we recommend `gpt-4o-mini`, as to date, it is the cheapest and fastest, though we also recommend to look up the prices periodically, as they are subject to change)
+- "API token": here you write your API token
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:75]:
```suggestion
- "API token": here you write your API token.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+
+## AI preferences
+
+Here are some new options in the JabRef preferences.
+
+![AI preferences](../img/AiPreferences.png)
+
+- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option if you want to use the new AI features
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face
+- "Chat model": choose the model you like (for OpenAI we recommend `gpt-4o-mini`, as to date, it is the cheapest and fastest, though we also recommend to look up the prices periodically, as they are subject to change)
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:74]:
```suggestion
- "Chat model": choose the model you like (for OpenAI we recommend `gpt-4o-mini`, as to date, it is the cheapest and fastest, though we also recommend to look up the prices periodically, as they are subject to change).
```

File: _posts/2024-08-16-AI-chatting.md [hunk 26]
```diff
@@ -0,0 +1,94 @@
...
+Please head to the [AI user documentation](https://docs.jabref.org/ai/ai-providers-and-api-keys) to learn about how to receive a key and where to enter it in the preferences.
+
+## AI preferences
+
+Here are some new options in the JabRef preferences.
+
+![AI preferences](../img/AiPreferences.png)
+
+- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option if you want to use the new AI features
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:73]:
```suggestion
- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 27]
```diff
@@ -0,0 +1,94 @@
...
+In order to use any service, you always need an API key.
+Please head to the [AI user documentation](https://docs.jabref.org/ai/ai-providers-and-api-keys) to learn about how to receive a key and where to enter it in the preferences.
+
+## AI preferences
+
+Here are some new options in the JabRef preferences.
+
+![AI preferences](../img/AiPreferences.png)
+
+- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option if you want to use the new AI features
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:72]:
```suggestion
- "Enable AI functionality in JabRef": by default it is turned off, so you need to check this option if you want to use the new AI features.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 28]
```diff
@@ -0,0 +1,94 @@
...
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face
+- "Chat model": choose the model you like (for OpenAI we recommend `gpt-4o-mini`, as to date, it is the cheapest and fastest, though we also recommend to look up the prices periodically, as they are subject to change)
+- "API token": here you write your API token
+- "Expert settings": the defaults provide good results. In case you would like to modify them, our [user documentation](https://docs.jabref.org/ai) provides an explanation of it
+
+## BONUS
+
+Have you ever considered running a local LLM on your computer while ensuring complete privacy?
+
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:81]:
```suggestion
```

File: _posts/2024-08-16-AI-chatting.md [hunk 29]
```diff
@@ -0,0 +1,94 @@
...
+We wrote a tutorial of how to install and use `ollama` and integrate it in JabRef.
+
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:83]:
```suggestion
``

File: _posts/2024-08-16-AI-chatting.md [hunk 30]
```diff
@@ -0,0 +1,94 @@
...
+## How does this work?
+
+In the background, JabRef analyses the linked PDF files of library entries. After indexing, this information is supplied to the AI, which, in our case, is a Large Language Model (LLM).
+The LLM is currently not stored on your computer. Instead, we have many integrations with AI providers (OpenAI, Mistral AI, Hugging Face), so you can choose the one you like the most.
+We send chunks of text to AI service and then receive processed responses. To use it, you need to configure JabRef with your API key.
+
+## Which AI provider should I use?
+
+For now, we recommend you choosing [OpenAI](https://platform.openai.com/docs/models).
+
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:57]:
```suggestion
```

File: _posts/2024-08-16-AI-chatting.md [hunk 31]
```diff
@@ -0,0 +1,94 @@
...
+For Mistral AI, you might need to make a subscription, whereas for OpenAI, a one-time payment option is available
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:58]:
```suggestion
For Mistral AI, you might need to make a subscription, whereas for OpenAI, a one-time payment option is available.
```

File: _posts/2024-08-16-AI-chatting.md [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
+
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:59]:
```suggestion
```

**Review by koppor — CHANGES_REQUESTED** [review]
One sentence per line, not per paragraph ^^.

I tried to polish some more.

File: _posts/2024-08-16-AI-chatting.md [hunk 33]
```diff
@@ -0,0 +1,102 @@
...
+- "AI provider": you can choose either OpenAI, Mistral AI, or Hugging Face.
+- "Chat model": choose the model you like
+- "API token": here you write your API token.
+- "Expert settings": the defaults provide good results. In case you would like to modify them, our [user documentation](https://docs.jabref.org/ai) provides an explanation of it.
+
+![AI preferences](../img/AiPreferences.png)
+
+## BONUS
+
+Have you ever considered running a local LLM on your computer while ensuring complete privacy?
```
Comment (ThiloteE) [inline _posts/2024-08-16-AI-chatting.md:92]: If you are considering running a local LLM on your computer to ensure complete privacy, check out my tutorial ...

**Comment by ThiloteE** [comment]
I took the liberty to rewrite a lot. You may see this as my part of the "review"

File: _posts/2024-08-16-AI-chatting.md [hunk 34]
```diff
@@ -0,0 +1,102 @@
+---
+title: "Introducing AI Features in JabRef"
+tags: [ai, gsoc]
+author: "[Ruslan](https://github.com/InAnYan)"
+---
+
+My name is Ruslan. I’m a novice contributor to JabRef, currently working on an AI project for [Google Summer of Code](https://summerofcode.withgoogle.com/).
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:7]:
```suggestion
My name is Ruslan and currently working on integrating AI into JabRef in the context of [Google Summer of Code 2024](https://summerofcode.withgoogle.com/).
```

File: _posts/2024-08-16-AI-chatting.md [hunk 35]
```diff
@@ -0,0 +1,102 @@
...
+
+In the window below, you can see the following elements:
+
+- The chat history with your messages
+- A button called "submit" that allows you to send a message (also commonly called "prompting")
+- A button for clearing the chat history
+
+![AI chat tab screenshot](../img/AiChat.png)
+
+
```
Comment (koppor) [inline _posts/2024-08-16-AI-chatting.md:36]: No double empty lines

```suggestion
```

**Comment by InAnYan** [comment]
I like the rewriting :)

File: _posts/2024-08-16-AI-chatting.md [hunk 36]
```diff
@@ -4,7 +4,7 @@ tags: [ai, gsoc]
 author: "[Ruslan](https://github.com/InAnYan)"
 ---
 
-My name is Ruslan. I’m a novice contributor to JabRef, currently working on an AI project for [Google Summer of Code](https://summerofcode.withgoogle.com/).
+My name is Ruslan and currently working on integrating AI into JabRef in the context of [Google Summer of Code 2024](https://summerofcode.withgoogle.com/).
```
Comment (Siedlerchr) [inline _posts/2024-08-16-AI-chatting.md:7]:
```suggestion
My name is Ruslan and I am currently working on integrating AI into JabRef in the context of [Google Summer of Code 2024](https://summerofcode.withgoogle.com/).
```

**Review by koppor — APPROVED** [review]

**Review by ThiloteE — APPROVED** [review]

**Review by koppor — APPROVED** [review]

**Comment by koppor** [comment]
Mistake from my side - used the console wrong. Need to reopen.
